### PR TITLE
docs: modernize the testing tutorial

### PR DESCRIPTION
**Pull request by dunglas**

### CODE REVIEW

File: distribution/testing.md [hunk 1]
```diff
@@ -69,17 +67,20 @@ The list of available generators as well as a cookbook explaining how to create
 
 ## Writing Functional Tests
 
-Now that you have some data fixtures for your API, you are ready to write functional tests with [PHPUnit](https://phpunit.de/index.html).
+Now that you have some data fixtures for your API, you are ready to write functional tests with [PHPUnit](https://phpunit.de).
+
+The API Platform test client implements the interfaces of [the Symfony HttpClient](https://symfony.com/doc/current/components/http_client.html). HttpClient is shipped with the API Platform distribution. The [the Symfony test pack](https://github.com/symfony/test-pack/blob/main/composer.json), which includes PHPUnit as well as Symfony components useful for testing, is also included.
```
Comment (vincentchalamon) [inline distribution/testing.md:72]:
```suggestion
The API Platform test client implements the interfaces of the [Symfony HttpClient](https://symfony.com/doc/current/components/http_client.html). HttpClient is shipped with the API Platform distribution. The [Symfony test pack](https://github.com/symfony/test-pack/blob/main/composer.json), which includes PHPUnit as well as Symfony components useful for testing, is also included.
```

Comment (vincentchalamon) [reply distribution/testing.md:72]: > The Symfony test pack (...) is also included [in the API Platform distribution].

No, it's not: https://github.com/api-platform/api-platform/blob/main/api/composer.json

**Review by vincentchalamon — CHANGES_REQUESTED** [review]

File: distribution/testing.md [hunk 2]
```diff
@@ -245,31 +246,35 @@ Check out the [testing documentation](../core/testing.md) to discover the full r
 ## Writing Unit Tests
 
 In addition to integration tests written using the helpers provided by `ApiTestCase`, all the classes of your project should be covered by [unit tests](https://en.wikipedia.org/wiki/Unit_testing).
-To do so, learn how to write unit tests with [PHPUnit](https://phpunit.de/index.html) and [its Symfony/API Platform integration](https://symfony.com/doc/current/testing.html).
+To do so, learn how to write unit tests with [PHPUnit](https://phpunit.de/) and [its Symfony/API Platform integration](https://symfony.com/doc/current/testing.html).
+
+## Continuous Integration, Continuous Delivery and Continuous Deployment
+
+Running your test suite in your [CI/CD pipeline](https://en.wikipedia.org/wiki/Continuous_integration) is important to ensure good quality and delivrery time.
```
Comment (alanpoulain) [inline distribution/testing.md:253]:
```suggestion
Running your test suite in your [CI/CD pipeline](https://en.wikipedia.org/wiki/Continuous_integration) is important to ensure good quality and delivery time.
```

File: distribution/testing.md (hunk 1 above)
Comment (dunglas) [reply distribution/testing.md:72]: It is :) https://github.com/api-platform/api-platform/blob/main/api/composer.json#L33

Packs are now automatically "unpacked" during installation.

**Review by vincentchalamon — APPROVED** [review]

**Pull request merged**